lib/standard/fixints.nit

```diff
@@ -0,0 +1,976 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# This file is free software, which comes along with NIT.  This software is
+# distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without  even  the implied warranty of  MERCHANTABILITY or  FITNESS FOR A
+# PARTICULAR PURPOSE.  You can modify it is you want,  provided this header
+# is kept unaltered, and a notification of the changes is added.
+# You  are  allowed  to  redistribute it and sell it, alone or is a part of
+# another product.
+
+# Basic integers of fixed-precision
+#
+# All classes defined here have C-equivalents and the semantics of their
+# operations are the same as C's
+#
+# * Int8 => int8_t
+# * Int16 => int16_t
+# * UInt16 => uint16_t
+# * Int32 => int32_t
+# * UInt32 => uint32_t
+#
+# NOTE: No UInt8 is provided as Byte is the same
+# SEE: kernel::Byte
+#
+# HOW TO USE:
+# All classes can be instanciated via a literal rule.
+# Namely, a suffix to append after the literal integer.
+#
+# * Int8 => i8
+# * Byte => u8
+# * Int16 => i16
+# * UInt16 => u16
+# * Int32 => i32
+# * UInt32 => u32
+module fixints
+
+import text
+
+redef class Numeric
+
+	# The Int8 equivalent of `self`
+	#
+	#     assert (-1).to_i8 == 0xFFFFi8
+	#     assert (1.9).to_i8 == 1i8
+	fun to_i8: Int8 do return self.to_i.to_i8
+
+	# The Int16 equivalent of `self`
+	#
+	#     assert (-1).to_i16 == 0xFFFFi16
+	#     assert (1.9).to_i16 == 1i16
+	fun to_i16: Int16 do return self.to_i.to_i16
+
+	# The UInt16 equivalent of `self`
+	#
+	#     assert (-1).to_u16 == 0xFFFFu16
+	#     assert (1.9).to_u16 == 1u16
+	fun to_u16: UInt16 do return self.to_i.to_u16
+
+	# The Int32 equivalent of `self`
+	#
+	#     assert (-1).to_i32 == 0xFFFFFFFFi32
+	#     assert (1.9).to_i32 == 1i32
+	fun to_i32: Int32 do return self.to_i.to_i32
+
+	# The UInt32 equivalent of `self`
+	#
+	#     assert (-1).to_u32 == 0xFFFFFFFFu32
+	#     assert (1.9).to_u32 == 1u32
+	fun to_u32: UInt32 do return self.to_i.to_u32
+
```

Comment:
Is there any kind of specification or documentation that can be linked to help users understand what is the behavior here?